File: sheet 8 Z. Decoding/main.py
def decode_polycarp_encoding(size, word):
    result = ""
    i = 0
    while size > 0:
        if size % 2 == 0:
            result = word[i] + result
        else:
            result = result + word[i]
        i += 1
        size -= 1
    return result

# Read input
size = int(input())
word = input().strip()

# Decode the encoded message
decoded_word = decode_polycarp_encoding(size, word)
print(decoded_word)

File: sheet 8 Z. Decoding/main.js
function decodePolycarpEncoding(n, s) {
    let result = "";
    let i = 0;
    
    while (n > 0) {
        if (n % 2 === 0) {
            result = s[i] + result;
        } else {
            result = result + s[i];
        }
        i++;
        n--;
    }
    
    return result;
}

// Read input
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let inputLines = [];
rl.on('line', (line) => {
    inputLines.push(line);
    if (inputLines.length === 2) {
        const n = parseInt(inputLines[0]);
        const s = inputLines[1];
        const decodedWord = decodePolycarpEncoding(n, s);
        console.log(decodedWord);
        rl.close();
    }
});
